dockerclient.go

```diff
@@ -212,6 +212,50 @@ func (client *DockerClient) ContainerChanges(id string) ([]*ContainerChanges, er
 	return changes, nil
 }
 
+func (client *DockerClient) readJSONStream(stream io.ReadCloser, decode func(*json.Decoder) decodingResult) (<-chan decodingResult, chan<- struct{}) {
+	resultChan := make(chan decodingResult)
+	closeChan := make(chan struct{})
+	go func() {
+		defer close(resultChan)
+
+		internalResultsChan := make(chan decodingResult)
+		defer close(internalResultsChan)
+
+		stillListening := make(chan struct{})
+		defer close(stillListening)
+
+		go func() {
```

Comment:
sure, I can move the nested goroutine out, and I will just move some of the variable initializations to the outside as well

the current code is actually buggy and this doesn't quite work with my fixed version which I'm about to push